Animview overflow

TheCycoONE opened the pull request:
*Fixes #1811*

Resolves undefined behavior by shifting unsigned longs.

clang-tidy suggested removing the unused assignment to pTmp, and fixing the name _merge (underscore names are reserved). I moved to using anonymous namespaces for file local code, per the c++ guidelines / style in CorsixTH

Alberth289346 commented:
I have a rule for myself not to assign to variables inside expressions, like
``` cpp
pFiles->Add(pTmp = new wxButton(this, ID_NEXT, L"Next"), 0,
              wxALIGN_CENTER | wxALL, 1);
```
It may have been necessary a few decades ago to save some fraction of time, but even then I preferred readability above being fast. I trust compilers to pull these tricks for me if feasible.

Don't know how you feel about it, but it could be something to fix.

----
The other thing was the question whether `1ul` is the right solution, since mask is a `std::uint32_t` and these two types are not always the same. To me a `static_cast` to the type of `mask` would have been more logical/safe, but you even removed that.

C++ standard seems to say `ul` suffix is at least 32 bit wide, so your solution is correct too.


So all seems fine to me.

TheCycoONE commented:
Split out all the other assignment's within Add spots I saw because I agree it's a bit ugly.
I removed setting the background colour to the button colour entirely. It had no effect on my laptop (gtk3) and I can't imagine it being good anywhere?
I also fixed a bug with the format strings and size_t types that caused errors for me when picking frames with > and <.

Alberth289346 commented on AnimView/frmMain.cpp:
```diff
@@ -314,61 +313,61 @@ frmMain::frmMain()
+  pFlags1->Add(m_chkFrameFlags[4], 0, wxEXPAND | wxALL, 2);
+  pFlags1->Add(m_chkFrameFlags[5], 0, wxEXPAND | wxALL, 2);
+  pFlags1->Add(m_chkFrameFlags[6], 0, wxEXPAND | wxALL, 2);
+  pFlags1->Add(m_chkFrameFlags[7], 0, wxEXPAND | wxALL, 2);
```
These array assignments start to look like it could use a loop, although I can also live without it.

TheCycoONE replied:
Yeah, I'll live with it unrolled, they aren't that long. Two sets of 8 flags / checkboxes.

Alberth289346 commented:
Changes look fine to me.
Hacked the cpp file somewhat as well, as I wanted to look at some animations, but the build was broken.
https://github.com/Alberth289346/CorsixTH/commit/c92abf97f2f0acc07d5d4f6dc5836e47944047f4

Lines 129 fixes a signed/unsigned comparison warning at line 130.
Then I started converting `unsigned char` to `std:uint8_t` and modifying casts.

The h file also fixes a comparison warning.

Feel free to apply or copy or redo what looks useful to you.

Alberth289346 commented:
And also updated, can't make a PR it seems.

https://github.com/CorsixTH/CorsixTH/compare/ac5c317c...Alberth289346:pr1813-warn_fixes-type_cleanup
